sjsonnet/src/sjsonnet/Val.scala

```diff
@@ -16,14 +16,27 @@ import scala.reflect.ClassTag
   * the tree can contain functions.
   */
 sealed abstract class Val {
+  def pos: Position
   def prettyName: String
+
   def cast[T: ClassTag: PrettyNamed] =
     if (implicitly[ClassTag[T]].runtimeClass.isInstance(this)) this.asInstanceOf[T]
     else throw new Error.Delegate(
       "Expected " + implicitly[PrettyNamed[T]].s + ", found " + prettyName
     )
-  def pos: Position
+
+  private[this] def failAs(err: String): Nothing =
```

Comment:
Should we make this take an implicit `T: PrettyNamed` for the error message, rather than passing in the `err` string manually? That would help ensure that we're consistent with our names for the various `Val.*` classes